### PR TITLE
Change plugin to use /google.golang.org/protobuf/compiler/protogen package

### PR DESCRIPTION
**Pull request by ncrypthic**
# Summary

Previous implementation to generate grpc method fullname code was using golang template. This PR changed the implementation to use google provided [protogen](/google.golang.org/protobuf/compiler/protogen) package. The package itself contain many useful functionality to help creating code generated file.

## Implementation detail

Current implementation using protogen.Plugin object provided by the package. The object also carries the original protobuf parsed file that can be used to iterate the protobuf syntax definitions. By using the parsed syntax definitions, we iterate each `service` & `rpc` definition in the object to generate our code using the provided `generator.P` function to simulate printing out our code to the plugin output buffer.

## Testing

Currently the test method is comparing the plugin output buffer content against the expected result of code generated.

### CODE REVIEW

**Comment by nicovogelaar** [comment]
LGTM. Thanks for upgrading @ncrypthic !

File: internal/test/test.pb.descriptor [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+�
+
+test.protoexamplegoogle/protobuf/empty.proto
google/protobuf/wrappers.proto2�
+
TestService;
+abc
.google.protobuf.StringValue.google.protobuf.Empty;
+Abc
.google.protobuf.StringValue.google.protobuf.EmptyB=Z;github.com/ncrypthic/protoc-gen-gofullmethods/internal/testbproto3
```
Comment (nicovogelaar) [inline internal/test/test.pb.descriptor]: Do we need to change the username `ncrypthic` here?

File: internal/test/test.proto [hunk 2]
```diff
@@ -0,0 +1,13 @@
+syntax="proto3";
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+
+package example;
+
+option go_package="github.com/ncrypthic/protoc-gen-gofullmethods/internal/test";
```
Comment (nicovogelaar) [inline internal/test/test.proto]: Do we need to change the username `ncrypthic` here?

File: example/server/server_test.go [hunk 3]
```diff
@@ -37,7 +38,7 @@ func TestMain(m *testing.M) {
 	}()
 	greeterClient = example.NewGreeterClient(greeterConn)
 
-	os.Exit(m.Run())
+	defer os.Exit(m.Run())
```
Comment (nicovogelaar) [inline example/server/server_test.go]: Is this `defer` useful here? Considering it is already the last statement of the function.

File: internal/generator.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import (
+	"fmt"
+	"strings"
+
+	"google.golang.org/protobuf/compiler/protogen"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/pluginpb"
+)
+
+const version = "1.0.0"
```
Comment (nicovogelaar) [inline internal/generator.go]: Would be nice if we can somehow override this with the current git tag / commit hash.

File: internal/test/test.pb.descriptor (hunk 1 above)
Comment (ncrypthic) [reply internal/test/test.pb.descriptor]: Yeah, will update it. I think we need to change this to `nicovogelaar` to be consistent with the package name.

File: internal/test/test.proto (hunk 2 above)
Comment (ncrypthic) [reply internal/test/test.proto]: will do :+1:

File: example/server/server_test.go (hunk 3 above)
Comment (ncrypthic) [reply example/server/server_test.go]: Right, we don't need `defer` here.

File: internal/generator.go (hunk 4 above)
Comment (ncrypthic) [reply internal/generator.go]: Never done this before, but I suppose we could use [runtime/debug.BuildInfo](https://pkg.go.dev/runtime/debug#BuildInfo) to do this. Let me see if I can change this.

**Review by nicovogelaar — APPROVED** [review]
LGTM. Thanks! @ncrypthic

**Comment by ncrypthic** [comment]
You're welcome @nicovogelaar . If you don't have anymore concern, I think we can merge and maybe add tag version for this @nicovogelaar ? I don't have permission to merge the PR.

**Comment by nicovogelaar** [comment]
All looks good @ncrypthic . I've made you a contributor now. Feel free to merge.

**Comment by ncrypthic** [comment]
Thanks @nicovogelaar! will tag it as version 0.2.0.

**Pull request merged**